File: sdks/php/src/Resources/StocktakesResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Stocktake;
use Puxbay\Models\PaginatedResponse;

class StocktakesResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("stocktakes/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => Stocktake::fromArray($item));
    }

    public function getStocktake(string $stocktakeId): Stocktake
    {
        $data = $this->get("stocktakes/{$stocktakeId}/");
        return Stocktake::fromArray($data);
    }

    public function create(Stocktake $stocktake): Stocktake
    {
        $data = $this->post('stocktakes/', $stocktake->toArray());
        return Stocktake::fromArray($data);
    }

    public function complete(string $stocktakeId): Stocktake
    {
        $data = $this->post("stocktakes/{$stocktakeId}/complete/");
        return Stocktake::fromArray($data);
    }
}

File: sdks/dotnet/src/Resources/StocktakesResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class StocktakesResource
    {
        private readonly PuxbayClient _client;
        public StocktakesResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Stocktake>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<Stocktake>>($"stocktakes/?page={page}");
        }

        public async Task<Stocktake> GetAsync(string stocktakeId)
        {
            return await _client.GetAsync<Stocktake>($"stocktakes/{stocktakeId}/");
        }

        public async Task<Stocktake> CreateAsync(Stocktake stocktake)
        {
            return await _client.PostAsync<Stocktake>("stocktakes/", stocktake);
        }

        public async Task<Stocktake> CompleteAsync(string stocktakeId)
        {
            return await _client.PostAsync<Stocktake>($"stocktakes/{stocktakeId}/complete/", null);
        }
    }
}
